core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -264,7 +264,21 @@ public Transaction(
      */
     @Override
     public Sha256Hash getHash() {
-        if (hash == null) {
+        return getHash(false);
+    }
+
+    public Sha256Hash getHash(boolean segwit) {
+        if (segwit) {
+            if (isCoinBase()) return Sha256Hash.ZERO_HASH;
+            ByteArrayOutputStream stream = new UnsafeByteArrayOutputStream(length < 32 ? 32 : length + 32);
+            try {
+                bitcoinSerializeToStream(stream, TransactionOptions.WITNESS);
+            } catch (IOException e) {
+                // Cannot happen, we are serializing to a memory stream.
```

Comment:
When serializing to a memory stream `IOException` cannot occur. Nonetheless the code for `bitcoinSerializeToStream` is generic enough to be able to take a stream that may serialize to disk or network, where `IOException` can happen.